src/app.rs

```diff
@@ -186,6 +199,35 @@ impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
         self
     }
 
+    /// Automatically registers components, adds resources and the rendering system.
+    pub fn with_renderer(self, pipe: PipelineBuilder) -> Result<Self> {
+        use cgmath::Deg;
+        use renderer::{Camera, Projection};
+        use ecs::components::{LightComponent, MaterialComponent, MeshComponent, Transform};
+        use ecs::resources::Factory;
+        use ecs::systems::RenderSystem;
+
+        let cam = Camera {
+            eye: [0.0, 0.0, -4.0].into(),
+            proj: Projection::perspective(1.3, Deg(60.0)).into(),
+            forward: [0.0, 0.0, 1.0].into(),
+            right: [1.0, 0.0, 0.0].into(),
+            up: [0.0, 1.0, 0.0].into(),
+        };
+
+        let render_sys = RenderSystem::new(&self.events, pipe)?;
+
+        let this = self.add_resource(cam)
+            .add_resource(Factory::new())
+            .register::<LightComponent>()
```

Comment:
Those `register` calls are fine while `SystemExt` isn't there.
Good to have them in one place at least.